odl/solvers/functional/functional.py

```diff
@@ -48,23 +49,33 @@ class Functional(Operator):
     <http://odlgroup.github.io/odl/guide/in_depth/functional_guide.html>`_.
     """
 
-    def __init__(self, space, linear=False, grad_lipschitz=np.nan):
+    def __init__(self, domain, linear=False, grad_lipschitz=np.nan,
```

Comment:
Should we change the order to `(self, domain, range, ...)`?
-  This might lead to incompatibilities with existing code like `Functional(space, False)`. Yes, I know that code like this is considered to be bad practise by many, but that probably doesn't stop users from doing so...
+  On the other hand it is the order of operator arguments (and more intuitive if you ask me).
-  For the `default_functionals` it can't be done since most have non-optional other arguments.

Reply:
> Should we change the order to (self, domain, range, ...)?

IMO definitely. Mostly because `Operator` works the same way, and I agree that it's the natural order of arguments.



> This might lead to incompatibilities with existing code like Functional(space, False). Yes, I know that code like this is considered to be bad practise by many, but that probably doesn't stop users from doing so...

Well I haven't seen any direct subclassing of `Functional` out there in the wild, so I wouldn't be too concerned with breaking shitty code like this.



> For the default_functionals it can't be done since most have non-optional other arguments.

True, but I feel this should be the special case, not the rule.